source/client.c

```diff
@@ -116,6 +120,19 @@ static void s_mqtt_client_shutdown(
     enum aws_mqtt_client_connection_state prev_state;
     { /* BEGIN CRITICAL SECTION */
         mqtt_connection_lock_synced_data(connection);
+        AWS_LOGF_TRACE(AWS_LS_MQTT_CLIENT, "id=%p: Lock hold from s_mqtt_client_shutdown 1", (void *)connection);
```

Comment:
could do something like this to keep your print statements, but not flood everyone
```
#define DEBUG_DEADLOCKS 0
#ifdef DEBUG_DEADLOCKS
#define LOGF_DEADLOCK_TRACE ....
```